pdc/apps/release/serializers.py

```diff
@@ -134,18 +135,32 @@ class Meta:
                   'compose_set', 'integrated_with', 'sigkey', 'allow_buildroot_push',
                   'allowed_debuginfo_services', 'allowed_push_targets')
 
+    def set_default_sigkey(self, validated_data):
+        try:
+            if "sigkey" in validated_data and not validated_data["sigkey"]:
+                validated_data["sigkey"] = SigKey.objects.get(key_id=settings.RELEASE_DEFAULT_SIGKEY)
+        except AttributeError:
+            # If there is no Settings.RELEASE_DEFAULT_SIGKEY, raise it in server log
+            raise AttributeError('Settings.RELEASE_DEFAULT_SIGKEY does not exist, please set one with the Sigkey key_id.')
```

Comment:
In a comment you say:

> User can still insert the data without sigkey, so nothing needs change in ct's input data or scripts.

but it doesn't seem to be possible here.

Can you also add tests for that?

Note: You can override settings in Django [like this](https://docs.djangoproject.com/en/1.8/topics/testing/tools/#overriding-settings).

Reply:
This try/catch just checks if the default key includes in db. 
And it means if people not insert one sigkey, will set a default one, so the users' scripts don't need change, they can still add the release without a sigkey input.

Reply:
But it requires `RELEASE_DEFAULT_SIGKEY` option to be set and refer to an existing sigkey. Right? This will require manual update of PDC server's configuration and data after upgrading to a new version.

Also, I don't understand where the default values ("redhatrelease2", "fd431d51") come from. Any specific data shouldn't be pushed to this repository.

Reply:
Yes, the default sigkey should be included into the db first.
And the default sigkey shouldn't change once it set even doing the upgrade.  This  ("redhatrelease2", "fd431d51") is set as default in ET's settings. And also this is already includes in the ET's database.  So we should keep the data same as ET's.

Reply:
What about Fedora? Those values should be in ansible repo; `pdc/settings_local.py.dist` is just template for the configuration file, ideally, it should also serve as documentation/reference for the options.

Reply:
aha, good question, I forgot one thing, if we close the pr about the sigkey name's changes, we can let SigKey.get_cached_id do this.